tests/basictests/trustyai_biased_models.sh

```diff
@@ -162,50 +154,39 @@ function teardown_trustyai_test() {
 
   REQUEST_ID="$(curl http://$TRUSTY_ROUTE/metrics/spd/requests | jq '.requests [0].id')"
 
-  os::cmd::expect_success_and_text "curl -X DELETE --location http://$TRUSTY_ROUTE/metrics/spd/request \
-    -H 'Content-Type: application/json' \
-    -d '{
-          \"requestId\": \"'"$REQUEST_ID"'\"
-        }'" "Removed"
-
-    os::cmd::expect_success_and_text "curl -X DELETE --location http://$TRUSTY_ROUTE/metrics/dir/request \
-      -H 'Content-Type: application/json' \
-      -d '{
-            \"requestId\": \"'"$REQUEST_ID"'\"
-          }'" "Removed"
+  # delete all requests
+  for METRIC_NAME in "spd" "dir"
+  do
+    for REQUEST in $(curl -s http://$TRUSTY_ROUTE/metrics/$METRIC_NAME/requests | jq -r '.requests [].id')
+    do
+      echo -n $REQUEST": "
+      curl -X DELETE --location http://$TRUSTY_ROUTE/metrics/$METRIC_NAME/request \
+          -H 'Content-Type: application/json' \
+          -d "{
+                \"requestId\": \"$REQUEST\"
+              }"
+      echo
+    done
+  done
 
   oc project $MM_NAMESPACE
-  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/common/secret.yaml"
-  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/biased_models/ovms-1.x.yaml"
-  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/biased_models/model0_onnx.yaml"
-  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/biased_models/model1_onnx.yaml"
+  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/trustyai/secret.yaml"
+  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/trustyai/ovms-1.x.yaml"
+  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/models/model0_onnx.yaml"
+  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/models/model1_onnx.yaml"
   os::cmd::expect_success "oc delete project $MM_NAMESPACE"
 }
 
-function teardown_test() {
-  oc project $ODHPROJECT
-  os::cmd::expect_success "oc delete -f ${RESOURCEDIR}/common/odh-core.yaml"
-  os::cmd::expect_success "oc delete project $ODHPROJECT"
-}
-
-
-if [ ! -z $1 ] && [ $1 == "clean" ]; then
-  teardown_trustyai_test
-elif [ ! -z $1 ] && [ $1 == "reset" ]; then
-  teardown_trustyai_test
-  teardown_test
-else
-  setup_test
-  check_trustyai_resources
-  deploy_model
-  check_mm_resources
-  check_communication
-  send_data
-  schedule_and_check_request
-  send_more_data
-  test_prometheus_scraping
-fi
-
 
 
+get_authentication
+check_trustyai_resources
+deploy_model
+check_mm_resources
+check_communication
+send_data
+schedule_and_check_request
+test_prometheus_scraping
+teardown_trustyai_test
```

Comment:
Are the clusters used for CI where this is running stateful? For example what does it happen if there is an error and the teardown procedure fails (or it is skipped)?
If yes, I would try to make more fault tolerant the code, for example, is it possible to move this clean up step in a different block (similar to always() condition in GH action) to make sure that if any of the previous steps has crash, the cluster is not going to remain in a not consistent state?

While if the cluster is recreated/cleaned up automatically after the job, this teardown should not be necessary